### ISSUE TITLE
[`Pylint`] `PLE2510` fix causes syntax error inside pre-3.12 f-strings

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ruff#18816]
### Summary

If a backspace is inside an f-string replacement while a pre-3.12 version is targeted, the fix for [invalid-character-backspace (PLE2510)](https://docs.astral.sh/ruff/rules/invalid-character-backspace/#invalid-character-backspace-ple2510) will introduce a syntax error. This happens both inside a string inside the replacement, and inside the format specs. [playground](https://play.ruff.rs/79ed94ae-5031-4d55-a53c-5a7fabe26700)
```
PS D:\python_projects> Get-Content issue.py
```
```py
f"{'␈'}"
```
^ note: I'm cheating here, PowerShell does not actually display the backspace character as `␈`, it actually gets displayed as `f"{'}"`
```
PS D:\python_projects> uvx ruff check issue.py --isolated --select PLE --target-version py39
```
```snap
issue.py:1:5: PLE2510 [*] Invalid unescaped character backspace, use "\b" instead
  |
1 | f"{'␈'}"
  |     ^ PLE2510
  |
  = help: Replace with escape sequence

Found 1 error.
[*] 1 fixable with the `--fix` option.
```
```
PS D:\python_projects> uvx ruff check issue.py --isolated --select PLE --target-version py39 --fix
```
```snap
issue.py:1:5: SyntaxError: Cannot use an escape sequence (backslash) in f-strings on Python 3.9 (syntax was added in Python 3.12)
  |
1 | f"{'\b'}"
  |     ^
  |

Found 2 errors (1 fixed, 1 remaining).
```

This also applies to `PLE2512`, `PLE2513`, and `PLE2515` [playground](https://play.ruff.rs/dc2bc095-7f43-4f4c-ba03-a1167e147196), as well as `PLE2514`, but the playground does not support null chars.

### Version

ruff 0.12.0 (87f0feb 2025-06-17) + playground

**Comment by MichaReiser** [comment]
This might be a bit annoying to fix but we can disable the fix if inside an interpolated element and the target version is older than 3.12

**Comment by LaBatata101** [comment]
> This might be a bit annoying to fix but we can disable the fix if inside an interpolated element and the target version is older than 3.12

The `PLE2510` check operates on the token stream, is there anyway of knowing if it's inside an interpolated element just with that information? The only way I could think of was getting the f-string text and using `parse_expression`.

**Comment by MichaReiser** [comment]
There are probably two options: 

1. Rewrite to an AST based rule
2. Look for `FSTRING_START` tokens and then ignore the middle tokens, https://play.ruff.rs/19ca2208-50b9-45ad-a41b-240b49eb2f6d

**Comment by omar-y-abdi** [comment]
I've been investigating this and have a fix ready. Submitting a PR shortly.

The fix tracks f-string/t-string interpolation nesting via a brace-depth stack in the token loop, then suppresses the autofix (but not the diagnostic) when the token is inside an interpolation and the target version doesn't support PEP 701 (< 3.12).
